src/vulkan-renderer/wrapper/make_info.cpp

```diff
@@ -151,6 +151,11 @@ VkInstanceCreateInfo make_info() {
     return ret;
 }
 
+template <>
+VkMemoryBarrier make_info() {
+    return {.sType = VK_STRUCTURE_TYPE_MEMORY_BARRIER};
```

Comment:
` .sType` hm?

Reply:
* In Vulkan, you need to set the so called `.sType` member of (almost) every struct.
* The technical reason for this is because the struct could be passed into some other code as a raw pointer, and the only way of identifying it is the `sType` member.
* You can read more about it here: https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#fundamentals-validusage-sType

We use the `make_info` templates to set up the sType automatically for us, making the code a little smaller.

Reply:
As an illustrative example, compare those equivalent code blocks:

```cpp
auto mem_barrier = make_info<VkMemoryBarrier>();
mem_barrier.srcAccessMask = VK_ACCESS_MEMORY_WRITE_BIT;
mem_barrier.dstAccessMask = VK_ACCESS_MEMORY_READ_BIT;
```

```cpp
VkMemoryBarrier mem_barrier;
mem_barrier.sType = VK_STRUCTURE_TYPE_MEMORY_BARRIER;
mem_barrier.srcAccessMask = VK_ACCESS_MEMORY_WRITE_BIT;
mem_barrier.dstAccessMask = VK_ACCESS_MEMORY_READ_BIT;
```

Reply:
Ah perfect thanks!